client/beefy/rpc/src/lib.rs

```diff
@@ -20,95 +20,78 @@
 
 #![warn(missing_docs)]
 
-use std::sync::Arc;
-
-use sp_runtime::traits::Block as BlockT;
-
-use futures::{FutureExt, SinkExt, StreamExt};
-use jsonrpc_derive::rpc;
-use jsonrpc_pubsub::{manager::SubscriptionManager, typed::Subscriber, SubscriptionId};
-use log::warn;
-
 use beefy_gadget::notification::BeefySignedCommitmentStream;
+use futures::{future, task::Spawn, FutureExt, StreamExt};
+use jsonrpsee::{
+	proc_macros::rpc,
+	types::{Error as JsonRpseeError, RpcResult},
+	SubscriptionSink,
+};
+use log::warn;
+use sc_rpc::SubscriptionTaskExecutor;
+use sp_runtime::traits::Block as BlockT;
 
 mod notification;
 
 /// Provides RPC methods for interacting with BEEFY.
-#[rpc]
+#[rpc(client, server, namespace = "beefy")]
```

Comment:
The client generates a `trait MyTraitClient` and impls it for the `T: jsonpsee_types::Client`

[Example here](https://github.com/paritytech/substrate/blob/jsonrpsee-polkadot-v0.9.12/bin/node/rpc-client/src/main.rs)

Reply:
I see. That's okay, but attempting to use the client will bring in the entire substrate dependency tree. We could try working around this with compilation features or spliiting the definitions to a separate crate. I think this might be addressed in follow up PRs though.

Reply:
@tomusdrw Would you prefer it if we had `#[rpc(server, namespace = "beefy")]` here instead? Do you actually use the Rust client anywhere?

I think there's a `jsonrpsee 0.2-beta` client in the lockfile still, coming from some bridges code. Not sure if that is relevant to this at all though.

Reply:
> Would you prefer it if we had #[rpc(server, namespace = "beefy")] here instead? Do you actually use the Rust client anywhere?

We would be using the client (even the one from `jsonrpc`) if it supported `async/await`, currently we define the client API manually (i.e. duplicate the code). After the switch it would be nice to use the substrate client definitions again to avoid duplication, but if that brings the entire dependency graph of substrate it is going to be prohibitevely too large so we will stick to the [manual client implementations](https://github.com/paritytech/parity-bridges-common/blob/6cd5775ebedf2544ef913639e8e5d40cd11f4286/relays/client-substrate/src/rpc.rs#L39).

> I think there's a jsonrpsee 0.2-beta client in the lockfile still, coming from some bridges code. Not sure if that is relevant to this at all though.

It's updated in bridges master, we are planning to bump the subtree soonish, so it will go away.